Organize settings with horizontal tabs

yourcontact opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

### Description
<!--- Describe your changes in detail -->

The idea is to move all Settings Titles to tabs at the top of the Settings form. The tabs will horizontally aligned. The user can click on tabs to go from one setting type to the other.

### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

This change is adding convenience to navigating Settings.

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->

WIP

### Screenshots (if appropriate):

WIP

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactoring

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project (run `$ yarn lint`).
<!---- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. -->

yourcontact commented:
@kytwb I aligned a couple of Settings Titles horizontally at the top of Settings form by adding `className="settings_titles"` to each of the `<h2>`

<img width="1701" alt="Screenshot 2020-04-13 at 18 51 21" src="https://user-images.githubusercontent.com/46404814/79152055-d5fa3880-7dcb-11ea-9a51-153bf27a736d.png">

```
{/* Titles */}
<div>
   <h2 id="general" className="settings_titles">{intl.formatMessage(messages.headlineGeneral)}</h2>
    &nbsp;&nbsp;
   <h2 id="apperance" className="settings_titles">{intl.formatMessage(messages.headlineAppearance)}</h2>
</div>
```

In "./Settings.scss", I added:
```
 .settings_titles {
    display:inline-block;
  }
```
Initially I wanted to add `className="settings_titles"` to the `<div>` but I could not make the CSS work to horizontally aligned the 2 titles.

@kytwb, let me know if you have any tips on styling CSS before I move forward.

kytwb commented:
Thanks for sharing your current progress via the draft pull request, appreciate it 🙏 You seem to have achieve what you want on the screenshot; I would not focus too much on the CSS for now as long as the core feature isn't finished as to keep the scope of what need to be addressed focused.

Same as you did for _Appearance_, I would add _Language_, _Advanced_ and _Updates_ so all the settings titles are displayed at the top and can be used as tabs. To use them as tabs, you'd need to conditionally display the right settings (below the tabs), based on which tab/settings title is clicked.

For that, you will have to [handle the `onClick` event](https://reactjs.org/docs/handling-events.html) on the settings titles, and store a value in a variable to reflect which tab has been clicked. Based on that variable value, you should then conditionally display settings blocks below the settings tabs/titles.

Happy to jump in a call to assist if needed! 🙏

kytwb commented:
@vantezzen can you confirm this is something worth continuing exploring given the scroll size of the current settings screen? This came out from one of the call we had with @yourcontact without going through an issue first, apologies for that.

vantezzen commented:
Yes, I think this is a great idea. I'd suggest using the tab layout used in the "Available Services" screen to keep a consistent design - WDYT?

yourcontact commented:
> I'd suggest using the tab layout used in the "Available Services" screen to keep a consistent design - WDYT?

Sharing a screenshot to make sure we are all on the page with regards to layout of settings tabs:

<img width="1701" alt="Screenshot 2020-04-14 at 22 02 00" src="https://user-images.githubusercontent.com/46404814/79268841-b7fc0900-7e9b-11ea-94bd-b0665ef09763.png">

cc @kytwb @vantezzen

kytwb commented:
LGTM 👍

vantezzen commented:
Yes, exactly what I was thinking about.

kytwb commented:
We are doing a pairing session to unlock the progress here with @yourcontact tomorrow 🙏

yourcontact commented:
![Kapture 2020-04-20 at 21 11 54](https://user-images.githubusercontent.com/46404814/79790170-a0b19580-834b-11ea-8fb6-62c13f8147e6.gif)

@kytwb, I started from scratch.
Now the state and onClick work.
The only piece of code I removed from when we were trying during the pairing sessions is this: 
```
const { activeSettingsTab } = this.state;
```
And I added back `this.state` to all `activeSettingsTab` on each settings tab in `className`:
```
className={this.state.activeSetttingsTab === 'general' ? 'badge badge--primary' : 'badge'}
```

The next step is for me to add conditions to the `div`s of each settings section to render based on whether the setting tab is active. I will take a look later this week.

Thank you again for the pairing session 🙏

yourcontact commented:
I added conditions `this.state.activeSetttingsTab === '[name of setting]' &&` on each setting section (e.g. "Advanced") to only show when its respective tab is active. It looks like this: 

![Kapture 2020-04-21 at 9 56 04](https://user-images.githubusercontent.com/46404814/79840606-6a5d3000-83b6-11ea-8f7a-d119c08bfbd8.gif)

Next steps are for me to:
- add spaces between tabs using CSS. For now I inserted `&nbsp;&nbsp;` between each tab. Without the `&nbsp;&nbsp;` the tabs look like this:

<img width="486" alt="Screenshot 2020-04-21 at 09 58 55" src="https://user-images.githubusercontent.com/46404814/79840899-dd66a680-83b6-11ea-8032-c1d7b1aaaa66.png">

- remove the space above the tabs:

<img width="1701" alt="Screenshot 2020-04-21 at 10 00 57 1" src="https://user-images.githubusercontent.com/46404814/79841000-0ab35480-83b7-11ea-83e5-23a5411c0221.png">

- other fixes?

Let me know. cc @kytwb

kytwb commented:
Looks good to me after the UI fine tuning, well done!

yourcontact commented:
I tweaked the CSS on a couple of files to add space in-between tabs https://github.com/getferdi/ferdi/pull/569/commits/a72aae6c4465abb8fb077e3fbcbd6c7085018c72 and remove space above tabs https://github.com/getferdi/ferdi/pull/569/commits/a72aae6c4465abb8fb077e3fbcbd6c7085018c72.

Before I removed some margin at the top https://github.com/getferdi/ferdi/pull/569/commits/a72aae6c4465abb8fb077e3fbcbd6c7085018c72 :

<img width="1701" alt="Screenshot 2020-04-22 at 12 50 47" src="https://user-images.githubusercontent.com/46404814/79976820-88ea2680-849d-11ea-9dec-2e1939b230b9.png">

After:

<img width="1701" alt="Screenshot 2020-04-22 at 12 51 35" src="https://user-images.githubusercontent.com/46404814/79976868-9ef7e700-849d-11ea-99c4-2fa2cacdfb03.png">

I was not too sure how much space to leave so I looked at spaces in "Available Services" and "Your Services" as a benchmark. Let me know I should do otherwise.

![Kapture 2020-04-22 at 12 50 20](https://user-images.githubusercontent.com/46404814/79977010-da92b100-849d-11ea-90b8-c8490fe7c106.gif)

kytwb commented on src/styles/type.scss:
```diff
@@ -19,7 +19,7 @@ h2 {
   letter-spacing: -1px;
   margin-bottom: 25px;
-  margin-top: 55px;
+  margin-top: 5px;
```
This change the `margin-top` on all the `h2` of the app. I'm afraid it might have undesired side effects, as in it might have changed the appearance of others `h2` used in the app?

yourcontact replied:
I checked but I could not see any, although you're right. Thank you for checking.

I am trying something like this:

<img width="343" alt="Screenshot 2020-04-22 at 15 11 16" src="https://user-images.githubusercontent.com/46404814/79985876-87bff600-84ab-11ea-96ea-2a98d9893891.png">

Is that how we should do it? @kytwb

kytwb commented on src/components/settings/settings/EditSettingsForm.js:
```diff
@@ -203,6 +213,7 @@ export default @observer class EditSettingsForm extends Component {
       isTodosActivated,
     } = this.props;
     const { intl } = this.context;
+    /* const { activeSettingsTab } = this.state; */
```
You can remove or use this, as you please.

yourcontact replied:
Resolved by https://github.com/getferdi/ferdi/pull/569/commits/cc52d3eeccf1bca1b38e9c7397fcf1d235c4e9ec

kytwb commented:
Looks good to me, need to confirm it still works as expected by trying and
if it does, that should do for now! 🙏 This is going to make the settings
much better!

yourcontact commented:
> Thank you @yourcontact! Awesome contribution! 💖

Thanks to you @kytwb for the pairing sessions!

yourcontact commented:
@kytwb, quick follow up from our off-github conversation - I checked locally on branch `refactor/setting-tabs`.  The margins I had added https://github.com/getferdi/ferdi/pull/569/commits/76f1d3fa90816252a3e149c4432d8aac284f035b stil works here locally, like in https://github.com/getferdi/ferdi/pull/569#issuecomment-617723130.

kytwb commented:
@yourcontact Thanks for the heads up. I saw that margin but it wasn't working anymore locally for me; re-addressed via https://github.com/getferdi/ferdi/pull/625/commits/3519d223906e6e8a73651734c65862f300384e7b. Sorry for the confusion, we will do a retrospective and enhance our process for the next iteration 🙏

yourcontact commented:
> @yourcontact Thanks for the heads up. I saw that margin but it wasn't working anymore locally for me; re-addressed via 3519d22. Sorry for the confusion, we will do a retrospective and enhance our process for the next iteration 🙏

Cool. No worries. Thank you, @kytwb. Looking forward for our next PR.
